packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -271,18 +271,52 @@ describeSpec('Listens:', [], () => {
         // (the document falls out) and send us a snapshot that's ahead of
         // docAv3 (which is already in our cache).
         .userListens(visibleQuery, 'resume-token-1000')
-        .watchAcksFull(visibleQuery, 5000, docAv2)
+        .watchAcks(visibleQuery)
+        .watchSends({ affects: [visibleQuery] }, docAv2)
+        .watchSnapshots(5000)
+        .watchSends({ affects: [visibleQuery] }, docAv3)
+        .watchCurrents(visibleQuery, 'resume-token-5000')
+        .watchSnapshots(6000)
         .expectEvents(visibleQuery, { fromCache: false })
         .userUnlistens(visibleQuery)
         .watchRemoves(visibleQuery)
         // Listen to allQuery again and make sure we still get docAv3.
         .userListens(allQuery, 'resume-token-4000')
         .expectEvents(allQuery, { added: [docAv3], fromCache: true })
-        .watchAcksFull(allQuery, 6000)
+        .watchAcksFull(allQuery, 7000)
         .expectEvents(allQuery, { fromCache: false })
     );
   });
 
+  specTest('Deleted documents in cache are fixed', ['exclusive'], () => {
+    const allQuery = Query.atPath(path('collection'));
+    const docAv1 = doc('collection/a', 1000, { key: 'a' });
+    const docDeleted = deletedDoc('collection/a', 2000);
+
+    return spec()
+      // Presuppose an initial state where the remote document cache has a
+      // broken synthesized delete at a timestamp later than the true version
+      // of the document.
+      .withGCEnabled(false)
+      .userListens(allQuery)
+      .watchAcksFull(allQuery, 1000, docAv1)
+      .expectEvents(allQuery, { added: [docAv1], fromCache: false })
+      .watchSends({ removed: [allQuery] }, docDeleted)
+      .watchCurrents(allQuery, 'resume-token-2000')
+      .watchSnapshots(2000)
```

Comment:
This can be `.watchSnapshots(2000, [allQuery], 'resume-token-2000')` and then you can get rid of the second `.watchCurrents()`.